### PR TITLE
Us900 Jail State Capture

### PR DESCRIPTION
**Pull request by kescob0825**
Context
A player can land on Go To Jail during any turn, including the first. The space resolution fires correctly, but the method responsible for setting the Player’s jail state on their data object was removed during the refactor, meaning the game has no way to actually flag a player as jailed. Without this, jail turn handling cannot be built or tested.

Description
The base PlayerController class listens for the JailStateChanged event raised by the Go To Launch Pad space or cards and correctly updates the player’s jail state, ending the turn immediately if they remain in jail.

Scope
Restore the method for setting jail state on the Player data object inside PlayerController
Sending a player to jail bypasses normal End Turn validation in TurnFlowCoordinator, since the turn ends as a result of space resolution rather than a player action (possible race condition on events)

### CODE REVIEW

File: Assets/Scripts/Managers/PlayerControllers/PlayerController.cs [hunk 1]
```diff
@@ -47,7 +50,8 @@ public PlayerController(
             UpgradeRequestEventChannel upgradeRequest,
             TurnActionRequestEventChannel turnActionRequest,
             TurnActionResultEventChannel  turnActionResult,
-            IntEventChannel bankruptPlayer)
+            IntEventChannel bankruptPlayer,
```
Comment (hdathert) [inline Assets/Scripts/Managers/PlayerControllers/PlayerController.cs:52]: these changes to the constructor (adding event channels) need to be propogated to the sub classes (`AIPlayerController` and `HumanPlayerController`), and then into the `PlayerManager` factory method, otherwise we will have compile errors.

**Review by hdathert — CHANGES_REQUESTED** [review]

**Review by hdathert — APPROVED** [review]
# Code Review Checklist

**Completed for**: US #900 and Tasks #924 to #925  
**Code Review Completed By**: @hdathert 

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*
